### PR TITLE
[Telink] Fix NotifyAdvertisingStopped with Null node

### PR DESCRIPTION
**Pull request by andriy-bilynskyy**
Since in file src/platform/Zephyr/BLEAdvertisingArbiter.cpp function InsertRequest
obtains previous node using iteration via slist -
condition (prev == nullptr) is only true if slist is empty.
In that case NotifyAdvertisingStopped is always called with Null
and no Null checker inside the function.
As result crash.

Testing
Tested manually with chip-tool.
Steps:
- Run: $ chip-tool pairing ble-thread <...>
- Wait till success

### CODE REVIEW

File: src/platform/Zephyr/BLEAdvertisingArbiter.cpp [hunk 1]
```diff
@@ -88,7 +88,6 @@ CHIP_ERROR InsertRequest(Request & request)
 
     if (prev == nullptr)
     {
-        NotifyAdvertisingStopped(sys_slist_peek_head(&sRequests));
```
Comment (bzbarsky-apple) [inline src/platform/Zephyr/BLEAdvertisingArbiter.cpp:91]: This does not seem right.  The goal of this code is that if we are inserting the new request at the head of the list, that means we are stopping advertising on the current list head.  

It does need to account for the "no current list head" case, but removing the code altogether seems like a strange way to do said accounting.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

**Review by s07641069 — APPROVED** [review]

**Comment by andriy-bilynskyy** [comment]
@bzbarsky-apple could you look at applied changes you mentioned?

File: src/platform/Zephyr/BLEAdvertisingArbiter.cpp [hunk 2]
```diff
@@ -108,6 +107,13 @@ CHIP_ERROR InsertRequest(Request & request)
 void CancelRequest(Request & request)
 {
     const bool isTopPriority = (sys_slist_peek_head(&sRequests) == &request);
+
+    // If cancelled request was top-priority, stop the advertising.
+    if (isTopPriority)
+    {
+        NotifyAdvertisingStopped(sys_slist_peek_head(&sRequests));
```
Comment (bzbarsky-apple) [inline src/platform/Zephyr/BLEAdvertisingArbiter.cpp:114]: I don't know whether this is right or not, (what's the API contract about notifying the consumer about advertising stop on a thing the consumer is canceling anyway?) but this does not address the question of what happend when InsertRequest inserts a new thing at the head....

**Comment by s07641069** [comment]
@andriy-bilynskyy Please close this PR as not valid.

**Pull request closed without merging**